presto-spark-base/src/main/java/com/facebook/presto/spark/execution/HttpNativeExecutionTaskInfoFetcher.java

```diff
@@ -83,7 +106,24 @@ public void onSuccess(BaseResponse<TaskInfo> result)
                             @Override
                             public void onFailure(Throwable t)
                             {
-                                log.error("TaskInfoCallback failed %s", t);
+                                // record failure
+                                try {
+                                    errorTracker.requestFailed(t);
+                                }
+                                catch (PrestoException e) {
+                                    stop();
+                                    lastException.set(e);
+                                    return;
+                                }
+                                ListenableFuture<?> errorRateLimit = errorTracker.acquireRequestPermit();
+                                try {
+                                    // synchronously wait on throttling
+                                    errorRateLimit.get(maxErrorDuration.toMillis(), TimeUnit.MILLISECONDS);
```

Comment:
We also ready pass maxErrorDuration when creating errorRateLimit. Why do we need to pass it here again?

Why are we waiting synchronously?